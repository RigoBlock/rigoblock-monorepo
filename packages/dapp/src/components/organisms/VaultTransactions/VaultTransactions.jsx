import './VaultTransactions.scss'
import { ETH_TO_WEI } from '../../../constants/utils'
import { connect } from 'react-redux'
import Icon from '../../atoms/Icon'
import Link from '../../atoms/Link'
import PropTypes from 'prop-types'
import React from 'react'
import Title from '../../atoms/Title'
import TransactionsTable from '../../molecules/TransactionsTable'
import moment from 'moment'

let VaultTransactions = ({ transactions, vaultAddress }) => {
  const linkComponent = link => (
    <Link to={link}>
      <Icon type="error_outline" className="link-icon" />
    </Link>
  )

  const vaultTransactions = transactions[vaultAddress]
  const parsedTransactions = vaultTransactions
    ? Object.keys(vaultTransactions).map(hash => ({
        ...vaultTransactions[hash],
        date: moment(vaultTransactions[hash].date).format('DD/MM/YY'),
        id: hash,
        transactionLink: linkComponent('#')
      }))
    : []

  const tableColumns = [
    {
      Header: 'ID',
      id: 'id',
      accessor: d => d.id.substr(d.id.length - 5, 5).toUpperCase(),
      maxWidth: 60
    },
    {
      Header: 'Date',
      id: 'date',
      accessor: d => moment(d.date).format('DD/MM/YY'),
      maxWidth: 70
    },
    {
      Header: 'Type',
      accessor: 'type',
      maxWidth: 75
    },
    {
      Header: 'Symbol',
      accessor: 'symbol',
      maxWidth: 60
    },
    {
      Header: 'Value',
      id: 'value',
      accessor: d => d.value.div(ETH_TO_WEI).toString(),
      maxWidth: 50
    },
    {
      Header: 'Units',
      id: 'units',
      accessor: d => d.units.toString(),
      maxWidth: 90
    },
    {
      Header: '',
      accessor: 'transactionLink',
      width: 40,
      sortable: false
    }
  ]

  return (
    <div className="vault-transactions">
      <Title>Transactions</Title>
      <TransactionsTable
        tableData={parsedTransactions}
        tableColumns={tableColumns}
      />
    </div>
  )
}

VaultTransactions.propTypes = {
  vaultAddress: PropTypes.string.isRequired,
  transactions: PropTypes.object
}

VaultTransactions.defaultProps = {
  transactions: {},
  columnWidths: []
}

VaultTransactions = connect(state => {
  const { currentAccount } = state.preferences
  return {
    transactions: currentAccount
      ? state.blockChain.accounts[currentAccount].vaultTransactions
      : {},
    location: state.routing.location.pathname
  }
})(VaultTransactions)

export default VaultTransactions
