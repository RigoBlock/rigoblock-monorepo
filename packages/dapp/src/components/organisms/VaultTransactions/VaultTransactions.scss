@import "../../_settings/variables.scss";

.vault-transactions {
  .title {
    padding-bottom: $dist-s;
  }
}
