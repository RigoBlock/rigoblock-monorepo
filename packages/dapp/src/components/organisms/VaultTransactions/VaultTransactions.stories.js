import '../../_settings/_base.scss'
import { BigNumber } from 'bignumber.js'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'
import React from 'react'
import VaultTransactions from './VaultTransactions'
import Web3 from 'web3'
import mockStore, { defaultState } from '../../../fixtures/store'

const transactionData = {
  date: 1531227188000,
  type: 'Deposit',
  symbol: 'ETH',
  value: new BigNumber('23860000000000000000'),
  units: new BigNumber('23860000'),
  account: '0x242b2dd21e7e1a2b2516d0a3a06b58e2d9bf9196'
}

const createTransaction = () => {
  const id = Web3.utils.randomHex()
  return {
    [id]: transactionData
  }
}

const getState = () => ({
  ...defaultState,
  ...{
    blockChain: {
      accounts: {
        '0x242B2Dd21e7E1a2b2516d0A3a06b58e2D9BF9196': {
          vaultTransactions: {
            '0x86a1ba4d485ce346bded508e2426798f825558be': {
              ...createTransaction(),
              ...createTransaction(),
              ...createTransaction(),
              ...createTransaction(),
              ...createTransaction(),
              ...createTransaction(),
              ...createTransaction(),
              ...createTransaction(),
              ...createTransaction(),
              ...createTransaction(),
              ...createTransaction(),
              ...createTransaction(),
              ...createTransaction(),
              ...createTransaction(),
              ...createTransaction(),
              ...createTransaction(),
              ...createTransaction(),
              ...createTransaction(),
              ...createTransaction(),
              ...createTransaction(),
              ...createTransaction(),
              ...createTransaction(),
              ...createTransaction(),
              ...createTransaction(),
              ...createTransaction(),
              ...createTransaction(),
              ...createTransaction(),
              '0xc8c775cec1666c7bd55eb60956dc2aadzxfw846396c051410838c74c6d37c': {
                date: 1531227188000,
                type: 'Withdraw',
                symbol: 'ETH',
                value: new BigNumber('1000000000000000000'),
                units: new BigNumber('1000000'),
                account: '0x242b2dd21e7e1a2b2516d0a3a06b58e2d9bf9196'
              },
              '0xc8c775cec1666asdasda0956dc2aadzxfw846396c051410838c74c6d37c': {
                date: 1531227188000,
                type: 'Withdraw',
                symbol: 'ETH',
                value: new BigNumber('900000000000000000000'),
                units: new BigNumber('999999999'),
                account: '0x242b2dd21e7e1a2b2516d0a3a06b58e2d9bf9196'
              }
            }
          }
        }
      }
    }
  }
})

const store = mockStore({ getState })

const VAULT_ADDRESS = '0x86a1ba4d485ce346bded508e2426798f825558be'

storiesOf('Organisms/VaultTransactions', module)
  .addDecorator(withKnobs)
  .addDecorator(story => <div style={{ width: '446px' }}>{story()}</div>)
  .addDecorator(story => <MemoryRouter>{story()}</MemoryRouter>)
  .addDecorator(story => <Provider store={store}>{story()}</Provider>)
  .add('default', () => <VaultTransactions vaultAddress={VAULT_ADDRESS} />)
