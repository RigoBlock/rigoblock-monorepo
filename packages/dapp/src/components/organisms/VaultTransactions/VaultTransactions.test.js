import { Provider } from 'react-redux'
import { shallow } from 'enzyme'
import React from 'react'
import VaultTransactions from './VaultTransactions.jsx'
import mockStore from '../../../fixtures/store'
import toJson from 'enzyme-to-json'

const store = mockStore({ mockFn: jest.fn })

const wrapper = shallow(
  <Provider store={store}>
    <VaultTransactions />
  </Provider>
)

describe('VaultTransactions component', () => {
  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
