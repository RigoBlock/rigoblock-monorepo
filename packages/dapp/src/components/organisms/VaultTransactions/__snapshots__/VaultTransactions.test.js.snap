// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VaultTransactions component renders correctly 1`] = `<Connect(VaultTransactions) />`;
